Support filtering by named queries in REST Data with Panache extension

Sgitario opened the pull request:
After https://github.com/quarkusio/quarkus/pull/29212 (filter by entity fields) is supported, we can now use namedQueries when filtering. 

With these changes, you can specify a named query to filter when listing the entities. For example, having the following named query in your entity:

```java
@Entity
@NamedQuery(name = "Person.containsInName", query = "from Person where name like CONCAT('%', CONCAT(:name, '%'))")
public class Person extends PanacheEntity {
  String name;
}
```

In this example, we have added a named query to list all the persons that contains some text in the `name` field. 

Next, we can set a query param `namedQuery` when listing the entities using the generated resource with the name of the named query that we want to use, for example, calling `http://localhost:8080/people?namedQuery=#Person.containsInName&name=ter` would return all the persons which name contains the text "ter".

geoand commented on docs/src/main/asciidoc/rest-data-panache.adoc:
```diff
@@ -405,6 +406,25 @@ Additionally, you can also filter by the entity fields by adding a query param w
+
+In this example, we have added a named query to list all the persons that contains some text in the `name` field. 
+
+Next, we can set a query param `namedQuery` when listing the entities using the generated resource with the name of the named query that we want to use, for example, calling `http://localhost:8080/people?namedQuery=#Person.containsInName&name=ter` would return all the persons which name contains the text "ter".
```
Is there a good reason why `#` is required for the named query name?

Sgitario replied:
It's required by Hibernate to use the named query: https://quarkus.io/guides/hibernate-reactive-panache#named-queries
However, I can add it programmatically.

geoand replied:
> However, I can add it programmatically.

I think that makes most sense, as I don't see any good reason for a user to be required to add this.

Sgitario replied:
Ok, it's done now. PR updated.

Sgitario committed:
Support filtering by named queries in REST Data with Panache extension

After https://github.com/quarkusio/quarkus/pull/29212 (filter by entity fields) is supported, we can now use namedQueries when filtering. 

With these changes, you can specify a named query to filter when listing the entities. For example, having the following named query in your entity:

```java
@Entity
@NamedQuery(name = "Person.containsInName", query = "from Person where name like CONCAT('%', CONCAT(:name, '%'))")
public class Person extends PanacheEntity {
  String name;
}
```

In this example, we have added a named query to list all the persons that contains some text in the `name` field. 

Next, we can set a query param `namedQuery` when listing the entities using the generated resource with the name of the named query that we want to use, for example, calling `http://localhost:8080/people?namedQuery=#Person.containsInName&name=ter` would return all the persons which name contains the text "ter".

geoand reviewed (approved):
Very nice!

Sgitario commented:
@geoand do you think this new feature along with https://github.com/quarkusio/quarkus/pull/29212, should be labeled with the `noteworthy-feature`? I think these are good candidates.

geoand commented:
Yeah, please add the label to one of the two and prepare a short sentence on what the feature is all about

Sgitario commented:
_For the release notes:_The REST Data with Panache extension now supports filtering by entity fields and using named queries.
